src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -0,0 +1,193 @@
+        if (menuImportLocalOpenApi == null) {
+            menuImportLocalOpenApi = new ZapMenuItem("openapi.topmenu.tools.importOpenApi");
+            menuImportLocalOpenApi.setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importOpenApi.tooltip"));
+
+            menuImportLocalOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    // Prompt for a OpenApi file.
+                    final JFileChooser chooser = new JFileChooser(Model.getSingleton().getOptionsParam().getUserDirectory());
+                    int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+                    if (rc == JFileChooser.APPROVE_OPTION) {
+                        extUrlOpenApiImport("file://" + chooser.getSelectedFile().getAbsolutePath());
+                    }
+
+                }
+            });
+        }
+        return menuImportLocalOpenApi;
+    }
+
+    /* Menu option to import a OpenApi file from a given URL. */
+    private ZapMenuItem getMenuImportUrlOpenApi() {
+        if (menuImportUrlOpenApi == null) {
+            menuImportUrlOpenApi = new ZapMenuItem("openapi.topmenu.tools.importRemoteOpenApi");
+            menuImportUrlOpenApi.setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importRemoteOpenApi.tooltip"));
+
+            final ExtensionOpenApi shadowCopy = this;
+            menuImportUrlOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    SwingUtilities.invokeLater(new Runnable() {
+
+                        @Override
+                        public void run() {
+                            new ImportFromUrlDialog(View.getSingleton().getMainFrame(), shadowCopy);
+                        }
+                    });
+                }
+            });
+        }
+        return menuImportUrlOpenApi;
+    }
+
+    public void extUrlOpenApiImport(final String url) { // TODO change url to defn or something...
+        Thread t = new Thread() {
+
+            @Override
+            public void run() {
+                this.setName(THREAD_PREFIX + threadId++);
+                Requestor requestor = new Requestor(HttpSender.MANUAL_REQUEST_INITIATOR);
+                try {
+                    requestor.addListener(new HistoryPersister());
+                    String defn = requestor.getResponseBody(url);
+                    Converter converter = new SwaggerConverter(defn);
+
+                    requestor.run(converter.getRequestModels());
+                } catch (SwaggerException e) {
+                    if (View.isInitialised()) {
+                        View.getSingleton().showWarningDialog(Constant.messages.getString("openapi.parse.error"));
+                    }
+                } catch (IOException e) {
```

Comment:
Should the exception be logged always? Otherwise there's no information on what caused the issue.